crates/bevy_reflect/src/list.rs

```diff
@@ -9,18 +9,41 @@ use crate::{
 
 /// An ordered, mutable list of [Reflect] items. This corresponds to types like [`std::vec::Vec`].
 ///
-/// This is a sub-trait of [`Array`] as it implements a [`push`](List::push) function, allowing
-/// it's internal size to grow.
+/// This is a sub-trait of [`Array`] as it implements [insertion](List::insert) and [removal](List::remove),
+/// allowing it's internal size to grow.
 ///
 /// This trait expects index 0 to contain the _front_ element.
 /// The _back_ element must refer to the element with the largest index.
 /// These two rules above should be upheld by manual implementors.
+///
+/// [`push`](List::push) and [`pop`](List::pop) have default implementations,
+/// however it may be faster to implement them manually.
 pub trait List: Reflect + Array {
+    /// Inserts an element at position `index` within the list.
```

Comment:
Nit: Should we include the rest of `Vec`'s [doc](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.insert) here? I know the elements might actually be shifted in memory, but their position within the list will reflect the shift.
```suggestion
    /// Inserts an element at position `index` within the list, shifting all elements after it to the right.
```

Reply:
i don't think the `List` trait has a concept of "to the right". `Vec`'s docs are explaining the change in memory layout where the bytes containing elements after the index are physically shifted to be to the right of the new element, whereas here we make no guarantee that's what is done to the memory.

perhaps we should have something more inline with [`VecDeque`'s docs](https://doc.rust-lang.org/stable/std/collections/struct.VecDeque.html#method.insert).

```suggestion
    // Inserts an element at position `index` within the list,
    // shifting all elements after it towards the back of the list.
```

Reply:
Good point. `VecDeque`'s docs are definitely more inline with the terminology we've been using for this trait.